openeo/rest/userfile.py

```diff
@@ -0,0 +1,71 @@
+import typing
+from typing import Any, Dict, Union
+from pathlib import Path
+from openeo.util import ensure_dir
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
+
+
+class UserFile:
+    """Represents a file in the user-workspace of openeo."""
+
+    def __init__(self, path: str, connection: 'Connection', metadata: Dict[str, Any] = None):
+        self.path = Path(path)
```

Comment:
based on you usage example screenshot (which shows this as `WindowsPath`), I now think this should be
```suggestion
        self.path = PurePosixPath(path)
```

because it is a non-concrete, posix-style path on the back-end

Reply:
I'm not sure however how well this plays with path construction operations elsewehere in this PR.

Reply:
Seems to work. I've made it so that PurePosixPath is only used for the back-end paths, but the download() methods still returns a path depending on the OS as it points to a locale file. But maybe also do a sanity check...